apps/web/src/components/safe-apps/AppFrame/SafeAppIframe.tsx

```diff
@@ -1,25 +1,38 @@
 import type { MutableRefObject, ReactElement } from 'react'
+import type { SafeAppDataWithPermissions } from '@/components/safe-apps/types'
 import css from './styles.module.css'
+import { sanitizeUrl } from '@/utils/url'
 
 type SafeAppIFrameProps = {
   appUrl: string
   allowedFeaturesList: string
   title?: string
   iframeRef?: MutableRefObject<HTMLIFrameElement | null>
   onLoad?: () => void
+  safeApp?: SafeAppDataWithPermissions
 }
 
 // see sandbox mdn docs for more details https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#attr-sandbox
 const IFRAME_SANDBOX_ALLOWED_FEATURES =
   'allow-scripts allow-same-origin allow-popups allow-popups-to-escape-sandbox allow-forms allow-downloads allow-orientation-lock'
 
-const SafeAppIframe = ({ appUrl, allowedFeaturesList, iframeRef, onLoad, title }: SafeAppIFrameProps): ReactElement => {
+const SafeAppIframe = ({
+  appUrl,
+  allowedFeaturesList,
+  iframeRef,
+  onLoad,
+  title,
+  safeApp,
+}: SafeAppIFrameProps): ReactElement => {
+  // Use the original URL with parameters if available, otherwise fallback to the provided URL
+  const safeAppUrl = safeApp?.originalUrl || appUrl
```

Comment:
A side note: this is becoming confusing. We should at some point streamline this logic and not pass `appUrl` as a prop at all + make `safeApp` always defined.